lib/colord/cd-test-shared.c

```diff
@@ -51,11 +52,15 @@ cd_test_get_filename (const gchar *filename)
 					 filename,
 					 NULL);
 	}
-	path = g_build_filename (TESTDATADIR, filename, NULL);
-	tmp = realpath (path, full_tmp);
-	if (tmp == NULL)
-		return NULL;
-	return g_strdup (full_tmp);
+	/* self tests are running */
+	testdatadir = g_getenv ("TESTDATADIR");
+	if (testdatadir != NULL) {
+		path = g_build_filename (testdatadir, filename, NULL);
```

Comment:
we can move the g_autoptr down here now...

Reply:
Like this?

Reply:
Looks good!